pandas/plotting/_matplotlib/core.py

```diff
@@ -1199,9 +1199,14 @@ def _kind(self) -> Literal["scatter"]:
 
     def __init__(self, data, x, y, s=None, c=None, **kwargs) -> None:
         if s is None:
-            # hide the matplotlib default for size, in case we want to change
-            # the handling of this argument later
-            s = 20
+            #  The default size of the elements in a scatter plot
+            # is now based on the rcParam ``lines.markersize``.
+            # This means that if rcParams are temporarily changed,
+            # the marker size changes as well according to mpl.rc_context().
+            if mpl.rcParams["_internal.classic_mode"]:
+                s = 20
```

Comment:
I think the default should still be 20 but deprecated in favor of using the `mpl.rcParams["lines.markersize"]`

Reply:
Hey, thanks for your review. Could you please clarify what needs to be fixed? Should I raise a deprecation warning if mpl.rcParams["_internal.classic_mode"] is True or remove this conditional altogether?

Reply:
The conditional should probably be something like

```
if s is None:
    warning.warning("The default of s=20 is deprecated and will change to mpl.rcParams["lines.markersize"]
                                 " in a future version. Specify `s` to suppress this warning",
                                 DeprecationWarning, stacklevel=find_stack_level())

```

Reply:
Hey @mroeschke , makes sense. Made the necessary changes.